runelite-client/src/main/java/net/runelite/client/plugins/banktags/tabs/TabInterface.java

```diff
@@ -720,6 +732,49 @@ private void deleteTab(String tag)
 		scrollTab(0);
 	}
 
+	private void renameTab(String oldTag)
+	{
+		chatboxPanelManager.openTextInput("Tag name")
+				.onDone((newTag) -> clientThread.invoke(() ->
+				{
+					if (!Strings.isNullOrEmpty(newTag) && !newTag.equalsIgnoreCase(oldTag))
+					{
+						if (tabManager.find(newTag) == null)
+						{
+							loadTab(newTag);
+							tabManager.move(oldTag, newTag);
+
+							TagTab oldTagTab = tabManager.find(oldTag);
+							TagTab newTagTab = tabManager.find(newTag);
+
+							if (oldTagTab != null && newTagTab != null)
+							{
+								Widget currentIcon = newTagTab.getIcon();
+								Widget newIcon = oldTagTab.getIcon();
+								currentIcon.setItemId(newIcon.getItemId());
+								newTagTab.setIcon(currentIcon);
+								newTagTab.setIconItemId(currentIcon.getItemId());
+								configManager.setConfiguration(CONFIG_GROUP, ICON_SEARCH + newTag, currentIcon.getItemId() + "");
+							}
+
+							deleteTab(oldTag);
+							tabManager.save();
+							tagManager.renameTag(oldTag, newTag);
+						}
+						else
+						{
+							chatboxPanelManager.openTextMenuInput("The specified bank tag already exists.")
+									.option("Ok", () ->
```

Comment:
Perhaps it would be good to give an option to merge into the existing tag too

Reply:
Added this along with removing the confirmation with 5347d76